greenlet.c

```diff
@@ -488,18 +506,20 @@ static int g_switchstack(void)
 		tstate->recursion_depth = target->recursion_depth;
 		tstate->frame = target->top_frame;
 		target->top_frame = NULL;
-#if PY_VERSION_HEX >= 0x030700A3
-		tstate->exc_state.exc_type = target->exc_type;
-		tstate->exc_state.exc_value = target->exc_value;
-		tstate->exc_state.exc_traceback = target->exc_traceback;
+#ifdef GREENLET_USE_EXC_INFO
+		tstate->exc_state = target->exc_state;
+		if (target->exc_info != NULL) {
```

Comment:
I think it might be better as a single line: `tstate->exc_info = target->exc_info ? target->exc_info : &tstate->exc_state`.

Reply:
done